cg/meta/workflow/analysis.py

```diff
@@ -135,6 +136,9 @@ def get_priority_for_case(self, case_id: str) -> int:
     def get_slurm_qos_for_case(self, case_id: str) -> str:
         """Get Quality of service (SLURM QOS) for the case."""
         priority: int = self.get_priority_for_case(case_id=case_id)
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        if are_all_samples_control(case=case):
+            priority = Priority.express
```

Comment:
We could consider changing the `get_priority_for_case` to take a `case` as input. Then we could fetch the `case` in this method once and just pass it to two methods. Seems like we are doing double work.

Reply:
I agree but I think it is used at some other places were you have only the `case_id` available. Also, I do not really want to change too many things in the code :/ .

Reply:
or just remove this function, it is only used there and it is one line of code. Worth considering

Reply:
Or much better:
```python
    def get_priority_for_case(self, case_id: str) -> int:
        """Get priority from the status db case priority"""
        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
        priority: int = case.priority or Priority.research
        if are_all_samples_control(case=case):
            priority = Priority.express
        return priority

    def get_slurm_qos_for_case(self, case_id: str) -> str:
        """Get Quality of service (SLURM QOS) for the case."""
        priority: int = self.get_priority_for_case(case_id=case_id)
        return Priority.priority_to_slurm_qos().get(priority)
```

Reply:
Relevant to link the discussion in the original issue: https://github.com/Clinical-Genomics/cg/issues/3768. I disagree with adding the `are_all_samples_control` to the `get_priority_for_case` since I do not think that the controls have anything to do with the cases priority. Given that there is a priority column on a case, I also expect any method claiming to get a case's priority to return the value in that column. 

I think it makes much more sense to have the `are_all_samples_control` in the `get_slurm_qos_for_case` since in my mind the sample controls should only affect the slurm qos, but not the case priority.

Reply:
Also, I only see `get_priority_for_case` used here - are you sure that it is used in other parts of the code @eliottBo ?

Reply:
I thought for some reasons you wanted to change `get_slurm_qos_for_case ` to take a Case. My bad.

Reply:
> Relevant to link the discussion in the original issue: #3768. I disagree with adding the `are_all_samples_control` to the `get_priority_for_case` since I do not think that the controls have anything to do with the cases priority. Given that there is a priority column on a case, I also expect any method claiming to get a case's priority to return the value in that column.
> 
> I think it makes much more sense to have the `are_all_samples_control` in the `get_slurm_qos_for_case` since in my mind the sample controls should only affect the slurm qos, but not the case priority.

In that case, I would say not handle changes in priority but changes directly in the priority_to_slurm_qos:
```python
    def get_slurm_qos_for_case(self, case_id: str) -> str:
        """Get Quality of service (SLURM QOS) for the case."""
        priority: int = self.get_priority_for_case(case_id=case_id)
        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
        slurm_qos: str = Priority.priority_to_slurm_qos().get(priority)
        if are_all_samples_control(case=case):
            slurm_qos: str = SlurmQos.EXPRESS
        return slurm_qos
```

Reply:
or, adding isaks suggestion:
```Python
def get_slurm_qos_for_case(self, case_id: str) -> str: 
    """Get Quality of service (SLURM QOS) for the case.""" 
    case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id) 
    priority: int = self.get_priority_for_case(case) 
    slurm_qos: str = Priority.priority_to_slurm_qos().get(priority) 
    if are_all_samples_control(case=case): 
        slurm_qos: str = SlurmQos.EXPRESS 
    return slurm_qos
```

Reply:
I like it! Teeny tiny change to be pedantic since the are_all_samples_control == True makes the rest of the checks useless:

```
def get_slurm_qos_for_case(self, case_id: str) -> SlurmQos: 
    """Get Quality of service (SLURM QOS) for the case.""" 
    case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
    if are_all_samples_control(case=case): 
        return SlurmQos.EXPRESS 
    priority: int = self.get_priority_for_case(case) 
    slurm_qos: SlurmQos = Priority.priority_to_slurm_qos().get(priority) 
    return slurm_qos
```

Reply:
FINAL VERSION!!! (sorry @eliottBo )
```python
def get_slurm_qos_for_case(self, case_id: str) -> SlurmQos: 
    """Get Quality of service (SLURM QOS) for the case.""" 
    case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
    if are_all_samples_control(case=case): 
        return SlurmQos.EXPRESS 
    priority: int = self.get_priority_for_case(case) 
    return Priority.priority_to_slurm_qos().get(priority) 
```